api/services/timelog.py

```diff
@@ -137,3 +139,73 @@ def check_task_for_overlap(self, task: Task) -> ValidatedObject:
                     f" {user_task_for_day.start_time} to {user_task_for_day.end_time}."
                 )
         return validated_task
+
+    def get_tasks_sum(self, user_id: int, start: date, end: date) -> int:
+        task_sum = (
+            self.db.query(func.sum(Task.task_total_minutes).label("task_sum"))
+            .filter(Task.user_id == user_id, Task.date.between(start, end))
+            .first()[0]
+        )
+        if task_sum is None:
+            task_sum = 0
+        return task_sum
+
+    def get_task_totals_projects(self, user_id: int, current_date: date) -> List[ProjectTaskSummary]:
+        totals = self.db.query(
+            func.public.get_user_project_summaries(user_id, current_date).table_valued(
+                "project_id", "project", "today_total", "today_text", "week_total", "week_text", "is_vacation"
+            )
+        ).all()
+        project_totals_list = []
+        for total in totals:
+            t = ProjectTaskSummary(
+                project_id=total[0],
+                project=total[1],
+                today_total=total[2],
+                today_text=total[3],
+                week_total=total[4],
+                week_text=total[5],
+                is_vacation=total[6],
+            )
+            if t.today_total is None:
+                t.today_total = 0
+            if t.week_total is None:
+                t.week_total = 0
+            project_totals_list.append(t)
```

Comment:
You can simplify the None checks in the assignment directly and even skip creating a variable for this:

```suggestion
            project_totals_list.append(
                ProjectTaskSummary(
                    project_id=total[0],
                    project=total[1],
                    today_total=total[2] or 0,
                    today_text=total[3],
                    week_total=total[4] or 0,
                    week_text=total[5],
                    is_vacation=total[6],
                )
            )
```

Reply:
Oh, that's so much better - thank you.